lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -131,49 +135,29 @@ defmodule KinesisClient.Stream.Shard.Producer do
   end
 
   @impl GenStage
-  def handle_info({:ack, _ref, [], failed_msgs}, state) do
-    Logger.debug("Retrying #{length(failed_msgs)} failed messages")
-
-    state =
-      case state.shard_closed_timer do
-        nil ->
-          state
-
-        timer ->
-          Process.cancel_timer(timer)
-          %{state | shard_closed_timer: nil}
-      end
-
-    {:noreply, failed_msgs, state}
+  def handle_info(msg, state) do
+    Logger.debug("ShardConsumer.Producer got an unhandled message #{inspect(msg)}")
+    {:noreply, [], state}
   end
 
-  @impl GenStage
-  def handle_info({:ack, _ref, successful_msgs, failed_msgs}, state) do
```

Comment:
So this case previously was for a mix of both failed and successful messages and we'd still checkpoint?

Reply:
> As noted, ideally we'd write our own Broadway.Acknowledger implementation (similar to how the Kafka adapter does it).

What's the LOE on this? Is it worth investing in now or the near future? If so, could we log a ticket to keep track of that work?

Reply:
I'll drop a ticket in here to track the work. I think it's beyond the LOE that I'm willing to put in now vs. getting back into extensions/datadog.